entities/cyclemanager/cyclecallbackgroup.go

```diff
@@ -205,7 +199,17 @@ func (c *cycleCallbackGroup) cycleCallbackParallel(shouldAbort ShouldAbortCallba
 				func() {
 					// cancel called in recover, regardless of panic occurred or not
 					defer c.recover(meta.customId, cancel)
-					executed := meta.cycleCallback(shouldAbort)
+					executed := meta.cycleCallback(func() bool {
+						if shouldAbort() {
+							return true
+						}
+
+						c.Lock()
+						defer c.Unlock()
```

Comment:
why we need a lock here?

Reply:
cause `meta.shouldAbort` can be changed in different routines on `deactivate` or `unregister`